app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -20,93 +23,80 @@ class Hawkular::MiddlewareManager < ManageIQ::Providers::MiddlewareManager
     has_many :middleware_deployments, :foreign_key => :ems_id, :dependent => :destroy
     has_many :middleware_datasources, :foreign_key => :ems_id, :dependent => :destroy
 
+    attr_accessor :client
+
     def verify_credentials(_auth_type = nil, options = {})
       begin
-
         # As the connect will only give a handle
         # we verify the credentials via an actual operation
-        connect(options).list_feeds
+        connect(options).inventory.list_feeds
       rescue => err
         raise MiqException::MiqInvalidCredentialsError, 'Invalid credentials'
       end
 
       true
     end
 
-    # Inventory
-
+    # Hawkular Client
     def self.raw_connect(hostname, port, username, password)
-      require 'hawkular_all'
-      url = URI::HTTP.build(:host => hostname, :port => port.to_i, :path => '/hawkular/inventory').to_s
-      ::Hawkular::Inventory::InventoryClient.new(url, :username => username, :password => password)
+      entrypoint = URI::HTTP.build(:host => hostname, :port => port.to_i).to_s
+      credentials = {
+        :username => username,
+        :password => password
+      }
+      options = {
+        :tenant => 'hawkular'
+      }
+      ::Hawkular::Client.new(:entrypoint => entrypoint, :credentials => credentials, :options => options)
     end
 
     def connect(_options = {})
-      self.class.raw_connect(hostname,
-                             port,
-                             authentication_userid('default'),
-                             authentication_password('default'))
+      @client ||= self.class.raw_connect(hostname,
+                                         port,
+                                         authentication_userid('default'),
+                                         authentication_password('default'))
     end
 
     def feeds
-      with_provider_connection(&:list_feeds)
-    end
-
-    def eaps(feed)
       with_provider_connection do |connection|
-        connection.list_resources_for_type(feed, 'WildFly Server', true)
+        connection.inventory.list_feeds
       end
     end
 
-    def child_resources(eap_parent)
+    def eaps(feed)
```

Comment:
Btw the `eaps` method is there since forever. We should not change it in this PR. The diff unfortunately makes it look like this is something brand new.